RELEASING.md

```diff
@@ -0,0 +1,69 @@
+# Release Checklist
+
+These steps assume that you've checked out the Socketto repository and are in the root directory of it.
+
+1. Ensure that everything you'd like to see released is on the `develop` branch.
+
+2. Create a release branch off `develop`, for example `jsdw-release-v0.6.0`. You'll want to decide how 
+   far the version needs to be bumped based on the changes to date. If unsure what to bump the version 
+   to (eg is it a major, minor or patch release), check with the Parity Tools team.
+
+2. Make sure that the tests pass (also checking that all examples compile).
+   
+   ```
+   cargo test
+   cargo test --examples
+   ```
+
+   If anything fails, it's likely worth fixing the failures in a PR and postponing the release. Minor 
+   issues can be fixed on this release branch.
+
+3. Check that you're happy with the current documentation.
+   
+   ```
+   cargo doc --open
+   ```
+ 
+   It's probably a good idea to check for broken links in the documentation, which
+   can be done using the third party tool `cargo-deadlinks`:
+
+   ```
+   cargo install cargo-deadlinks
```

Comment:
ok, is this the same as `RUSTDOCFLAGS="--deny broken_intra_doc_links" cargo doc --document-private-items`?

Reply:
I think we should enforce this in the CI instead.

Reply:
Oooh, I didn't realise that you could check internal links with `cargo doc`!

`cargo deadlinks` checks external links as well by making an HTTP request, but it is an extra thing to depend on.

I def agree about CI: I adapted the CI stuff from `jsonrpsee` here: https://github.com/paritytech/soketto/pull/40 (no cargo deadlinks, just the build-in check for now). If I can get that merged in a timely fashion I'll merge it in here and update the release docs accordingly. Otherwise, this next release will have to be "manual", and I can tweak the docs when the CI PR does merge :)